### PR TITLE
Tessera merge

### PR DESCRIPTION
**Pull request by jagpreetsinghsasan**
#185

### CODE REVIEW

File: platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl [hunk 1]
```diff
@@ -0,0 +1,79 @@
+apiVersion: flux.weave.works/v1beta1
```
Comment (sownak) [inline platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl]: is the helm_component.tpl empty file needed?

File: platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    git: {{ git_url }}
+    ref: {{ git_branch }}
+    path: {{ charts_dir }}/node_tessera  
+  values:
+    replicaCount: 1
+    metadata:
+      namespace: {{ component_ns }}
+      labels:
+    images:
+      node: quorumengineering/quorum:2.1.1
```
Comment (sownak) [inline platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl]: take the versions from network.yaml

File: platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      othernodes:
+{% for tm_node in network.config.tm_nodes %}
+        - url: {{ tm_node }}
+{% endfor %}
+      tls: {{ network.config.tm_tls | upper }}
+      trust: {{ network.config.tm_trust | upper }}
+    genesis: {{ genesis }}
+    staticnodes:
+{% for enode in enode_data_list %}
+      - enode://{{ enode.enodeval }}@{{ enode.peer_name }}.{{ external_url }}:{{ enode.p2p_ambassador }}?discport=0&raftport={{ enode.raft_ambassador }}
```
Comment (sownak) [inline platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl]: raftport should be dependent on consensus. IBFT does not need RAFT port

File: platforms/quorum/configuration/roles/setup/vault_kubernetes/tasks/main.yaml [hunk 4]
```diff
@@ -48,7 +48,7 @@
 - name: "Get Kubernetes cert files for organizations"
   shell: |
     KUBECONFIG={{ kubernetes.config_file }} kubectl -n {{ component_ns }} get secrets $(KUBECONFIG={{ kubernetes.config_file }} kubectl -n {{ component_ns }} get secrets | grep "default" | awk '{print $1}') -o jsonpath="{.data['ca\.crt']}" | base64 --decode > "./build/{{ component_ns }}.ca.cert"
-  when: auth_list.stdout.find(component_auth) == -1
+  # when: auth_list.stdout.find(component_auth) == -1
```
Comment (sownak) [inline platforms/quorum/configuration/roles/setup/vault_kubernetes/tasks/main.yaml]: why commented?

File: platforms/quorum/configuration/roles/setup/vault_kubernetes/tasks/main.yaml [hunk 5]
```diff
@@ -111,3 +111,11 @@
...
     VAULT_TOKEN: "{{ vault.root_token }}"
   when: auth_list.stdout.find(component_auth) == -1     # Run when auth_path is not found
+
+#####################################################################################################################
+# This task creates the docker pull credentials for image registry
+- name: Create the docker pull credentials
+  shell: |
+    KUBECONFIG={{ kubernetes.config_file }} kubectl create secret -n {{ component_ns }} docker-registry regcred --docker-server={{ network.docker.url }} --docker-username={{ network.docker.username }} --docker-password={{ network.docker.password }} --docker-email="test@abc.mail"
+  when: auth_list.stdout.find(component_auth) == -1
+  ignore_errors: yes
```
Comment (sownak) [inline platforms/quorum/configuration/roles/setup/vault_kubernetes/tasks/main.yaml]: newline?

File: platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    ref: {{ git_branch }}
+    path: {{ charts_dir }}/node_tessera  
+  values:
+    replicaCount: 1
+    metadata:
+      namespace: {{ component_ns }}
+      labels:
+    images:
+      node: quorumengineering/quorum:{{ network.version }}
+      alpineutils: adopblockchaincloud0502.azurecr.io/alpine-utils:1.0
```
Comment (sownak) [inline platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl]: previous comment applied to all the items here. even alpineutils should use the docker url from network.yaml
as well has tessera version

Comment (jagpreetsinghsasan) [reply platforms/quorum/configuration/roles/helm_component/templates/tessera.tpl]: Sure, I have done it now.

**Pull request merged**